### PR TITLE
Create vehicle controller

### PR DESCRIPTION
**Pull request by Prashanttiwari2190**

### CODE REVIEW

**Comment by Prashanttiwari2190** [comment]
@sumitkanugo Please review.

**Review by sumitkanugo — APPROVED** [review]

File: src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using ChromeData;
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs:6]: @Prashanttiwari2190  Don't Import this service here. It is meant to be only restricted to Data Layer.

File: src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        public UXGetMakeQueryHandler(IChromeDataService chromeDataService)
+        {
+            this.chromeDataService = chromeDataService;
+        }
+
+        public async Task<IEnumerable<Division>> Handle(UXGetMakeQuery request, CancellationToken cancellationToken)
+        {
+            var divisions = await chromeDataService.GetDivisions(request.Year);
+            return divisions;
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs:24]: @Prashanttiwari2190 Please Select only divisionId & divisionName in to custom created Mode Division, Which I already created under Model. We are not supposed to use Chrome Data Models. We will be using only our own created model as per our Project needs.

File: src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQuery.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿using ChromeData;
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQuery.cs:1]: Please remove this import from everywhere in Application Layer.

File: src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        public UXGetModelQueryHandler(IChromeDataService chromeDataService)
+        {
+            this.chromeDataService = chromeDataService;
+        }
+
+        public async Task<IEnumerable<Model>> Handle(UXGetModelQuery request, CancellationToken cancellationToken)
+        {
+            var models = await chromeDataService.GetModels(request.Year, request.Division);
+            return models;
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQueryHandler.cs:24]: @Prashanttiwari2190  Please return only ModelId & ModelName from Custom Model already created.

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+using SFM.VehicleBuilder.Domain.Correlation;
+
+namespace SFM.VehicleBuilder.Web.Controllers.V1
+{
+    /// <summary>
+    ///   A controller used to service Fleetio work orders.
+    /// </summary>
+    [ApiController]
+    [ApiVersion("1.0")]
+    [ApiExplorerSettings(GroupName = "Vehicle Builder")]
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:25]: Vehicle Builder is the project name, Please change this to Step 1 name called 'vehicle'. Also change API name to same, @Prashanttiwari2190

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        ///   Gets the model list.
+        /// </summary>
+        /// <param name="year"> year information.</param>
+        /// <param name="division"> division information.</param>
+        /// <returns>Returns a <see cref="string"/> containg the status of the model.</returns>
+        [ProducesResponseType(typeof(string), 200)]
+        [ProducesResponseType(typeof(string), 401)]
+        [ProducesResponseType(typeof(string), 404)]
+        [ProducesResponseType(typeof(string), 500)]
+        [HttpGet("{year}/{division}")]
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:92]: Please add api route as --> "{year}/model/{division}"

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        ///    Gets the Make list.
+        /// </summary>
+        /// <param name="year">year information.</param>
+        /// <returns>Returns a <see cref="string"/> containg the status of the division.</returns>
+        [ProducesResponseType(typeof(string), 200)]
+        [ProducesResponseType(typeof(string), 401)]
+        [ProducesResponseType(typeof(string), 404)]
+        [ProducesResponseType(typeof(string), 500)]
+        [HttpGet("{year}")]
+        public async Task<ActionResult<IEnumerable<Division>>> GetMake([FromRoute] int year)
```
Comment (abhinavatai) [inline src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:60]: Please chane api name as --> "make/{year}"

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 5 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:25]: Done

File: src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQueryHandler.cs (hunk 4 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQueryHandler.cs:24]: Done

File: src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQuery.cs (hunk 3 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Application/Queries/UXGetModelQuery/UXGetModelQuery.cs:1]: Done

File: src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs (hunk 2 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs:24]: Done

File: src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs (hunk 1 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Application/Queries/UXGetMakeQuery/UXGetMakeQueryHandler.cs:6]: Done

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 7 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:60]: Done.

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 6 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:92]: Done.

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 7 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:60]: done

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 6 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:92]: done

File: src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs (hunk 5 above)
Comment (Prashanttiwari2190) [reply src/SFM.VehicleBuilder.Web/Controllers/V1.0/UxVehicleBuilderController.cs:25]: done

**Pull request merged**